### ISSUE TITLE
Style/PreferredHashMethods did not find a has_key?

### ISSUE THREAD

**Issue by dennisvandehoef** [bbatsov/rubocop#3554]
--------

## Expected behavior

Finding the `has_key?` method, and willing to replace it with `key?`

## Actual behavior

Did not say anney thing about the `has_key?` method

## Steps to reproduce the problem

I did not change anything to the default config of this cop.

Just run robocop, this is the code that got checked

```ruby
  class Foo
      include ActionView::Helpers::TagHelper # content_tag
      include ActionView::Helpers::TextHelper # concat
      include ActionView::Helpers::FormOptionsHelper

      def initialize(form, value)
        @form = form
        @value = value
      end

      def method_missing(method, *args)
        value.fetch(method, '') || super
      end

      def respond_to_missing?(method_name, include_private = false)
        value.has_key?(method_name) or super
      end
  end
```

## RuboCop version

```
$ rubocop -V
0.43.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin14)
```

**Comment by savef** [comment]
This works fine for me with the default config, I get the following offence:
```
x.rb:16:13: C: Style/PreferredHashMethods: Use Hash#key? instead of Hash#has_key?.
      value.has_key?(method_name) or super
            ^^^^^^^^
```
This seems fine for me on the `v0.42.0` and `v0.43.0` tags and on `master`. Try passing the `--debug` switch to see where your config is being inherited from.

**Comment by dennisvandehoef** [comment]
```bash
be rubocop --debug app/dsls/widget_edit_form_creator/builder/base.rb

For /Users/dennisvandehoef/code/kp/cms: configuration from /Users/dennisvandehoef/code/kp/cms/.rubocop.yml
Inheriting configuration from /Users/dennisvandehoef/code/kp/cms/.rubocop_todo.yml
Default configuration from /Users/dennisvandehoef/.rvm/gems/ruby-2.3.1/gems/rubocop-0.43.0/config/default.yml
Inheriting configuration from /Users/dennisvandehoef/.rvm/gems/ruby-2.3.1/gems/rubocop-0.43.0/config/enabled.yml
Inheriting configuration from /Users/dennisvandehoef/.rvm/gems/ruby-2.3.1/gems/rubocop-0.43.0/config/disabled.yml
Inspecting 1 file
Scanning /Users/dennisvandehoef/code/kp/cms/app/dsls/widget_edit_form_creator/builder/base.rb
.

1 file inspected, no offenses detected
Finished in 0.686972149007488 seconds
```
.robocop_todo.yml
```yaml
# Offense count: 2
# Configuration parameters: CountComments.
Metrics/ClassLength:
  Exclude:
  - app/models/experiment.rb

# This needs to be fixed before we can update tu Ruby 2.4
# https://github.com/bbatsov/rubocop/pull/3492
Lint/UnifiedInteger:
  Exclude:
  - lib/core_extensions/fixnum.rb
  - spec/lib/publisher/obfuscator_spec.rb
```

.rubocop.yml
```yaml
# Longterm goal: Eliminate exemptions from this todo file
inherit_from: .rubocop_todo.yml

AllCops:
  TargetRubyVersion: 2.3
  Exclude:
    - db/**/*
    - config/**/*
    - bin/**/*
    - Rakefile
    - lib/generators/**/*
    - lib/tasks/spec.rake
    - playground/**/*
    - Gemfile

Documentation:
  Enabled: false
DotPosition:
  EnforcedStyle: 'leading'
AndOr:
  Enabled: false
Not:
  Enabled: false
SymbolProc:
  Enabled: false
ClassAndModuleChildren:
  Enabled: false
LineLength:
  Max: 79
Lint/UnneededDisable:
  Enabled: false
Style/AccessorMethodName:
  Exclude:
    - app/shortcodes/**/*
IfUnlessModifier:
  MaxLineLength: 79
Style/MultilineOperationIndentation:
  Enabled: false
Metrics/AbcSize:
  Max: 25
Metrics/ClassLength:
  Max: 150 # Don't increase!
Metrics/ModuleLength:
  Max: 120 # Don't increase!
Style/SpaceAroundOperators:
  AllowForAlignment:
    - '='
    - '=>'
    - '||='
    - '+='
Rails/TimeZone:
  Enabled: false
Rails/Date:
  Enabled: false
Style/FrozenStringLiteralComment:
  Enabled: false
Style/NumericPredicate:
  Enabled: false

# Enable in the > 0.43.0 version, it has to manny bugs now
Style/VariableNumber:
  Enabled: false

```
